### PR TITLE
[XLA GPU] fix pascal batched gemm numeric issues

### PR DESCRIPTION
**Pull request by Cjkkkk**
Fix the legacy cublas batched gemm failure at arch <= Pascal.

### CODE REVIEW

**Review by reedwm — CHANGES_REQUESTED** [review]
Good catch!

The merge commit has some gemm_rewrite tests disable checking the HLO on pre-Volta, such as [here](https://github.com/openxla/xla/blob/bd068e0f8954b63b91fa98df8b6ce6884eec0f5c/xla/service/gpu/tests/gemm_rewrite_test.cc#L1763-L1764). Can you remove this checks?

**Comment by Cjkkkk** [comment]
> Good catch!
> 
> The merge commit has some gemm_rewrite tests disable checking the HLO on pre-Volta, such as [here](https://github.com/openxla/xla/blob/bd068e0f8954b63b91fa98df8b6ce6884eec0f5c/xla/service/gpu/tests/gemm_rewrite_test.cc#L1763-L1764). Can you remove this checks?

Thanks, I forgot to remove these in the tests. Removed now.

**Review by reedwm — APPROVED** [review]

**Review by reedwm — APPROVED** [review]

File: xla/service/gpu/gemm_rewriter.cc
Comment (ddunl) [inline xla/service/gpu/gemm_rewriter.cc:1]: seems like `cuda_compute_capability_` is unused now, we should delete it I think

Comment (reedwm) [reply xla/service/gpu/gemm_rewriter.cc:1]: Yep, the internal linter is complaining about this so we cannot submit until it's deleted.

Comment (Cjkkkk) [reply xla/service/gpu/gemm_rewriter.cc:1]: thanks for the reminder, removed now.

**Comment by ddunl** [comment]
I think needs a rebase as well because I'm seeing some complaints about conflicts on the CI

**Comment by Cjkkkk** [comment]
> I think needs a rebase as well because I'm seeing some complaints about conflicts on the CI

rebased now. should be good.

**Review by reedwm — APPROVED** [review]

**Pull request closed without merging**